### ISSUE TITLE
Segfault when printing a slice

### ISSUE THREAD

**Issue by mhristache** [rust-lang/rust#66685]
This code is causing a segfault:

```rust
fn main() {
    let a = [0u8; std::u32::MAX as usize];
    print(&a);
}

fn print(s: &[u8]) {
    println!("{:?}", s);
}
```

[Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3b1f0f96eaca688adb042642ee934d07)

**Comment by nagisa** [comment]
It is not printing the slice that’s segfaulting, but the fact that you’re allocating a 4GB array on stack.

Amusingly stack probing fails to work here as the amount of bytes to probe for overflows and code ends up probing only a few bytes:

```
	movl	$88, %eax
	callq	__rust_probestack
```

**Comment by nagisa** [comment]
https://bugs.llvm.org/show_bug.cgi?id=44129 for probe size overflow.

**Comment by nagisa** [comment]
Inclined to close this bug as invalid, as there is nothing actionable on our end here. We cannot make OS allocate us an infinite stack for the main thread, sadly. Consider `stacker` or perhaps creating a thread with a larger stack.

**Comment by mhristache** [comment]
I don't question the fact that print shouldn't work in this case. 
The strange thing for me is that there seems to be some check that gets triggered if I try to print the slice in main() but not if the println! statement is added to a wrapper function, e.g. this doesn't compile (expected):

```rust
fn main() {
    let a = [0u8; std::u32::MAX as usize];
    println!("{:?}", &a);
    print(&a);
}

fn print(s: &[u8]) {
    println!("{:?}", s);
}
```

```
error[E0277]: arrays only have std trait implementations for lengths 0..=32
 --> src/main.rs:3:22
  |
3 |     println!("{:?}", &a);
  |                      ^^ the trait `std::array::LengthAtMost32` is not implemented for `[u8; 4294967295]`
  |
  = note: required because of the requirements on the impl of `std::fmt::Debug` for `[u8; 4294967295]`
  = note: required because of the requirements on the impl of `std::fmt::Debug` for `&[u8; 4294967295]`
  = note: required by `std::fmt::Debug::fmt`
```
